pkg/armrpc/asyncoperation/worker/registry.go

```diff
@@ -65,14 +70,46 @@ func (h *ControllerRegistry) Register(ctx context.Context, resourceType string,
 	return nil
 }
 
+// RegisterDefault registers a default controller that will be used when no other controller is found.
+//
+// The default controller will be used when Get is called with an operation type that has no registered controller.
+// The default controller will not be cached by the registry.
+func (h *ControllerRegistry) RegisterDefault(ctx context.Context, factoryFn ControllerFactoryFunc, opts ctrl.Options) error {
+	h.ctrlMapMu.Lock()
+	defer h.ctrlMapMu.Unlock()
+
+	h.defaultFactory = factoryFn
+	h.defaultOpts = opts
+	return nil
+}
+
 // Get gets the registered async controller instance.
-func (h *ControllerRegistry) Get(operationType v1.OperationType) ctrl.Controller {
+func (h *ControllerRegistry) Get(ctx context.Context, operationType v1.OperationType) (ctrl.Controller, error) {
 	h.ctrlMapMu.RLock()
 	defer h.ctrlMapMu.RUnlock()
 
 	if h, ok := h.ctrlMap[operationType.String()]; ok {
-		return h
+		return h, nil
 	}
 
-	return nil
+	return h.createDefault(ctx, operationType)
+}
+
+func (h *ControllerRegistry) createDefault(ctx context.Context, operationType v1.OperationType) (ctrl.Controller, error) {
+	if h.defaultFactory == nil {
+		return nil, nil
+	}
```

Comment:
Isn't this function more like **getDefault**? If **defaultFactory** is nil, then we return nil for the controller. So, to me, it is more like **getDefault** that is called from the **Get** function if no controller is registered.

Reply:
Is this feedback about the naming or behavior? I don't have a strong opinion about the name 😆

Reply:
It is about the name of the function not the behavior. It seems like it doesn't create the default but get the existing default.